### ISSUE TITLE
Support indexing ObjList while connecting to disk

### ISSUE THREAD

**Issue by kygx-legend** [husky-team/husky#256]

**Comment by kygx-legend** [comment]
@lmatz please help this issue for external memory.

**Comment by ddmbr** [comment]
This issue is a good one!

My idea for `ObjList` is that it's just an abstraction, allowing users to lookup and traverse. Underlying the interface it could be disk storage or anything. This should make many things much easier.

**Comment by lmatz** [comment]
Is it appropriate to make a buffer pool which handles read/write requests from ObjLists and AttrLists, then fetches the pages which are necessary from disk. Sort of like how database does its job. So it may handle the situation where multiple ObjLists or AttrLists are being processed in husky-h4.

**Comment by kygx-legend** [comment]
We have thought about that but the main problem is how to handle non-pod object in a ObjList. The object can be defined anything via template. The best solution is to transform it into a BinStream. Based on your idea, I think it could work if combining buffer pool with BinStream. It's more complicated to do such an abstraction, which means much time. We want some simple and direct implementation.

**Comment by kygx-legend** [comment]
@ddmbr yeah, your idea is great and similar with mine. We can start from this.

**Comment by kygx-legend** [comment]
See https://github.com/kygx-legend/husky/tree/em.
